ReactWindows/ReactNative.Net46/Bridge/DispatcherHelpers.cs

```diff
@@ -17,12 +18,47 @@ public static void AssertOnDispatcher()
 
         public static bool IsOnDispatcher()
         {
-            return Thread.CurrentThread == Dispatcher.CurrentDispatcher.Thread;
+            var dispatcher = Application.Current != null
+                ? Application.Current.Dispatcher
+                : Dispatcher.CurrentDispatcher;
+
```

Comment:
I have implemented a static member on the DispatcherHelpers class where the dispatcher must be explicitly set... The dispatcher helpers static methods then use the set value instead of assuming Application.Current.Dispatcher or Dispatcher.Current...